### ISSUE TITLE
Reimplement model list on BSP UseCases

### ISSUE THREAD

**Issue by antonelepfl** [cnr-ibf-pa/hbp-bsp-issues#360]
Just to put everyone on the same page I would like to clarify some stuffs about the model list on Hippocampus.

We have if I'm not mistaken 3 types of sets:

- **Hippocampus** - Hippocampus cells (previously in Luca's github repo and now in CSCS container)
- **SSCx** - SomatosensoryCortex
- **Striatal** interneuron - Alex Kozlov models

_Currently_ we have models lists on the following Use Cases:

- Morphology analysis:
   - Morphology analysis (with notebook) -> **Hippocampus**
   - Morphology visualization -> **Hippocampus** + **SSCx**
- Single cell building:
   - Rebuild an existing single hippocampal cell -> **Hippocampus**
   - Optimize a striatal fast-spiking interneuron -> **Striatal**
- Single cell in silico experiments:
   - Single cell in silico experiment under current clamp -> **Hippocampus** + **SSCx**

The _new implementation_ should add Stefano Masoli's models so we get something like:
- ...
- Single cell in silico experiments:
   - Single cell in silico experiment under current clamp -> **Hippocampus** + **SSCx** + _**Granule cells**_

@SteMasoli can you confirm that is the only place for your models?

Is there anything that should be added / changed regarding the models that appears on each list / use case?

**Comment by mmigliore** [comment]
Hi Stefano,

the list should also include "small circuit" and "brain area" in silico
experiments.
Cheers
Michele

**Comment by antonelepfl** [comment]
Ciao Michele,

In this list I was only talking about the models lists that we have. As you said, we have _Small Circuit In Silico Experiments_ and _Brain Area Circuit In Silico Experiments_ already as a nav items on the Online Use Cases.

**Comment by antonelepfl** [comment]
Hi @a1eko, Is your striatal fast-spiking interneuron available on the Model Catalog?

**Comment by a1eko** [comment]
FS cell used to be in BSP, what's the problem with it?

**Comment by SteMasoli** [comment]
@antonelepfl 
"can you confirm that is the only place for your models?"

Under "Single cell building:" there are the three usecases which are the sources of the BlueNaaS models.

"Optimize a cerebellar granule cell model composed by a single section" -> Cerebellum
"Optimize a cerebellar granule cell multicompartmental model" -> Cerebellum
"Simulation and validation of a mouse Purkinje cell multicompartmental model" -> Cerebellum

Since the BlueNaaS models are not directly produced by the usecases themselves, i think they will have to remain in two distinct sections. 

So, yes, they can be added to the:
Single cell in silico experiments:
    Single cell in silico experiment under current clamp -> Hippocampus + SSCx + Cerebellum

**Comment by antonelepfl** [comment]
I have implemented the list of models from the Model Catalog in the [DEV](https://collab.humanbrainproject.eu/#/collab/8444/nav/65635) version @SteMasoli @alex4200 please check if everything work as expected

I also wanted to point out that the Granule cell **does not** have any _Morphology_ image nor _Responses_ traces so please try to put them there so we have a uniform experience.

**Comment by alex4200** [comment]
In the DEV environment it seems to work to choose the two models from the list for 'Single Cell Experiment'. 

However, I cannot find the models in the 'Morphology Analysis'. I would have expected them here as well...

**Comment by antonelepfl** [comment]
As described in the description in Morphology Analaysis there are only the Hippocampus

**Comment by alex4200** [comment]
@antonelepfl  Ok my bad. However, I find that very confusing...

**Comment by antonelepfl** [comment]
It was confusing for me too that is why I created this ticket so we are all in the same page regarding to the models :)

**Comment by appukuttan-shailesh** [comment]
@antonelepfl regarding:
> I also wanted to point out that the Granule cell does not have any Morphology image nor Responses traces so please try to put them there so we have a uniform experience.

I should point out that the _Morphology_ and _Responses_ traces  are not mandatory for the model catalog. These have been provided by the Hippocampus teams and thus made available, but isn't as per any standard convention (atleast as of now) that is followed on the model catalog. If owners of other existing models want to follow the same, it is possible to add these by going to the model's page on the model catalog and using the "edit" feature and then using "Add an image".

**Comment by jdcourcol** [comment]
@appukuttan-shailesh  this issue is the user experience when browsing the list of selected models to execute a use case. If some models do not have the morphology image and response trace, then the GUI looks bad.
If the model catalog is the reference for what is being displayed in the use cases, I think it should enforce these standards to get a good user experience. I don't think it should be up to the end user good will to provide these image or not.
However, we should help these end users. If their models follows standards, I assume one can provide an executable that generates these images and store them along with the model

**Comment by alex4200** [comment]
I have created a section in the google-docs document as a reference of this information: [see here](https://docs.google.com/document/d/19Ni5fGFv6PTTM1D_D2FN4dj5IRhtfq1D0V2OWjXxJww/edit#heading=h.d8l6c6in37ti)

Let me know if you do not have access to this doc.

**Comment by appukuttan-shailesh** [comment]
@jdcourcol : I completely agree about the need for consistency in GUI appearance across all the models. What's happening here is that we are using a feature of the Model Catalog that was not intended for this kind of usage. And since we never had a formal discussion informing model owners of following such a convention, this problem is expected.

From the point of view of the model catalog, it might not be possible to mandate all entries to have a morphology and responses trace as it hosts a variety of models for which these might not be applicable. For starters, we could simply require that for models to be included as part of use cases, model owners follow a certain convention for specifying info on the model catalog. But even for this, it is best if we discuss, finalize and disseminate this info to everyone involved.

@apdavison : Do let us know if you have a different take on using the model catalog for such purposes.

**Comment by SteMasoli** [comment]
@antonelepfl 
> 
> I have implemented the list of models from the Model Catalog in the [DEV](https://collab.humanbrainproject.eu/#/collab/8444/nav/65635) version @SteMasoli @alex4200 please check if everything work as expected
> 
> I also wanted to point out that the Granule cell **does not** have any _Morphology_ image nor _Responses_ traces so please try to put them there so we have a uniform experience.

Thank you. 
Is it possible to add the Purkinje cell model too? In this way there will be an optimized model for each optimization usecase.

@appukuttan-shailesh 

>  If owners of other existing models want to follow the same, it is possible to add these by going to the model's page on the model catalog and using the "edit" feature and then selecting "Add an image".

Thank you. 
I will add a screenshot of the morphologies and one for the traces, like in the case of the Hippocampus. 
Is there a size limit for the images?

**Comment by antonelepfl** [comment]
@SteMasoli Just to clarify which models from the catalog do you want to add in the `Single cell in silico experiments`? 

I see these models:

![screenshot 2018-12-13 at 15 11 49](https://user-images.githubusercontent.com/24406041/49943966-83fc8680-fee9-11e8-95f9-cc79a40eb7aa.png)

You can retrieve them using:
```
curl -X GET \
  'https://validation-v1.brainsimulation.eu/models/?brain_region=Cerebellum&cell_type=Purkinje%20Cell&model_scope=Single%20cell%20model' \
  -H 'Postman-Token: 0fe1fff7-46e3-4a85-a2c0-752f0cd1dadf' \
  -H 'cache-control: no-cache'
```
Are those the selected to be on the Model List?

**Comment by appukuttan-shailesh** [comment]
@SteMasoli 
> Is there a size limit for the images?

Not sure if you intended the memory size or the display size of the image, so will answer for both ;)

**Memory size:** You need to upload these images online, and then specify the URL (to this image) on the model's page. So we don't actually host the images, and thus no restrictions on this size.

**Display size:** No such limitations have been imposed. If a large image is provided, we provide a smaller thumbnail on the model's page which can be clicked to view the full-size image.

**Comment by jdcourcol** [comment]
@alex4200 we should put that in the agenda of Monday meeting.
The issue will be that if we don't have a tool to generate these images they will look different for each model (colors, scale and so on..)

**Comment by alex4200** [comment]
@jdcourcol  Yes I agree. Because I included a link to this ticket in the email I sent around, it should be straightforward to talk about it next Monday.

**Comment by SteMasoli** [comment]
The Purkinje cell, on the model catalog, is defined with:

Name - SGA2 M9 
ID - c8d184e3-fb75-4ac4-a99d-8b864593f396 
Created on - 03/12/2018, 05:10

**Comment by antonelepfl** [comment]
I have just put the on DEV version
